File: docs/JAMIA/comparisons/RPython/TM/41815365_geom_rect_string_dates.R
# ID: SO#41815365
# Language: R
# Bug class: Value type
# Runtime consequence: Crash
# In study: Yes
# Inclusion rationale: date_trans requires Date class, got character. String where date expected.
# Reproduction status: Live
# Type system catch: `Temporal` rejects `string`

library(ggplot2)
library(scales)

Data <- data.frame(
  Date = c("2002-05-23", "2002-05-29", "2002-05-31"),
  Well = c("MW-3", "MW-3", "MW-3"),
  Elev = c(929.04, 929.39, 929.37),
  stringsAsFactors = FALSE
)
Data$Date <- as.Date(Data$Date, format = "%Y-%m-%d")
Data$Well <- as.factor(Data$Well)
Data$Elev <- as.numeric(Data$Elev)

ggplot(data = Data, aes(x = Date, y = Elev, group = Well, colour = Well)) +
  geom_line(linewidth = 0.75) +
  xlab("") +
  ylab("Elevation (ft.)") +
  scale_color_brewer(palette = "Spectral") +
  scale_x_date(
    breaks = date_breaks("2 year"),
    labels = date_format("%Y")
  ) +
  theme_bw() +
  geom_rect(
    data = Data,
    aes(
      xmin = "2004-04-29",
      xmax = "2004-12-20",
      ymin = -Inf,
      ymax = Inf
    ),
    fill = "gray",
    alpha = 0.5
  )

File: docs/JAMIA/comparisons/RPython/TM/41815365_geom_rect_string_dates.ts
/**
 * RPython SO#41815365 — geom_rect with character xmin/xmax on Date axis
 * Effect: Crash
 * Bug class: Type coercion
 *
 * R bug: ggplot2 date_trans scale requires Date objects for xmin/xmax in geom_rect.
 * Passing character strings crashes: "Invalid input: date_trans works with objects
 * of class Date only."
 *
 * In tidy-ts, date strings cannot be used where numeric/temporal operations are expected.
 * The type system catches the mismatch at the point of numeric consumption.
 */
import { createDataFrame, stats as s } from "@tidy-ts/dataframe";

const Data = createDataFrame([
  { date: "2002-05-23", well: "MW-3", elev: 929.04 },
  { date: "2002-05-29", well: "MW-3", elev: 929.39 },
  { date: "2002-06-15", well: "MW-1", elev: 930.12 },
]);

// The SO user's intent: plot elevation over time with date-based rectangles.
// ggplot2 needs Date objects; strings crash date_trans.
// tidy-ts equivalent: correlation of date strings with elevation requires numeric x.
// @ts-expect-error — string[] is not assignable to number[]
s.test.correlation.pearson({ x: Data.extract("date"), y: Data.extract("elev") });
